app/models/system.rb

```diff
@@ -9,7 +9,7 @@ class System < ApplicationRecord
 
   after_initialize :init
 
-  has_many :activations, dependent: :destroy
+  has_many :activations, dependent: :delete_all
```

Comment:
Please add a comment here that this is done for performance optimization, else you run the risk that somebody just changes it back in the future, not knowing why we apply this special behavior here.